pkg/server/restart.go

```diff
@@ -324,43 +351,53 @@ func loadSandbox(ctx context.Context, cntr containerd.Container) (sandboxstore.S
 	createdAt := info.CreatedAt
 
 	// Load sandbox status.
-	t, err := cntr.Task(ctx, nil)
-	if err != nil && !errdefs.IsNotFound(err) {
-		return sandbox, errors.Wrap(err, "failed to load task")
-	}
-	var s containerd.Status
-	var notFound bool
-	if errdefs.IsNotFound(err) {
-		// Task is not found.
-		notFound = true
-	} else {
-		// Task is found. Get task status.
-		s, err = t.Status(ctx)
-		if err != nil {
-			// It's still possible that task is deleted during this window.
-			if !errdefs.IsNotFound(err) {
-				return sandbox, errors.Wrap(err, "failed to get task status")
-			}
-			notFound = true
-		}
-	}
 	var state sandboxstore.State
 	var pid uint32
-	if notFound {
-		// Task does not exist, set sandbox state as NOTREADY.
-		state = sandboxstore.StateNotReady
-	} else {
-		if s.Status == containerd.Running {
-			// Task is running, set sandbox state as READY.
-			state = sandboxstore.StateReady
-			pid = t.Pid()
+	if err := func() error {
```

Comment:
I didn't change any logic, wrap the status generation logic in a function.

Reply:
ok but why?